### ISSUE TITLE
Duplicate KubeController-manager-arg  in kubecontroller manifest

### ISSUE THREAD

**Issue by seb-835** [rancher/rke2#3376]
**Environmental Info:**
RKE2 Version:  1.24.4+rke2r1
Node(s) CPU architecture, OS, and Version:  3.10.0-1160.71.1.el7.x86_64

**Describe the bug:**
i want to set arg  terminated-pod-gc-threshold to 500 for Kubecontroller
But got a duplicate entry in the manifest with default and new values

```
kubectl get pod -n kube-system kube-controller-manager-fr0 -o yaml --kubeconfig /etc/rancher/rke2/rke2.yaml | grep gc
    - --terminated-pod-gc-threshold=1000
    - --terminated-pod-gc-threshold=500
```

**Steps To Reproduce:**
i add this in  my rke2  config.file :
```
...
kube-controller-manager-arg:
  - "terminated-pod-gc-threshold=500"
 ```
and do : 
```
systemctl restart rke2-server.service
```
**Expected behavior:**
Only on entry in yaml:
```
kubectl get pod -n kube-system kube-controller-manager-fr0 -o yaml --kubeconfig /etc/rancher/rke2/rke2.yaml | grep gc
    - --terminated-pod-gc-threshold=500
```

Is it a bug or my bad ?

**Comment by brandond** [comment]
We just append the user-provided args to our defaults; we don't make any attempt to de-duplicate identical flags. The Kubernetes flag parser itself handles deduplicating or merging the values as necessary.

**Comment by seb-835** [comment]
Do you know how kubernetes deduplicate ?
Does it use the latest duplicate arg in the list ?

**Comment by brandond** [comment]
Yes, it takes the last one. That's why we put our defaults first, and the user provided args second.
